src/firefly/js/visualize/ui/MouseReadout.jsx

```diff
@@ -18,88 +18,255 @@ import {showMouseReadoutOptionDialog} from './MouseReadoutOptionPopups.jsx';
 import CoordinateSys from '../CoordSys.js';
 import CysConverter from '../CsysConverter.js';
 import CoordUtil from '../CoordUtil.js';
+import VisUtil from '../VisUtil.js';
+
 import debounce from 'lodash/debounce'; //TEST CODE
 import {callGetFileFlux} from '../../rpc/PlotServicesJson.js'; //TEST CODE
 
+
 var rS= {
-	border: '1px solid white',
-	width: 500,
+	//border: '1px solid white',
+	width: 550,
 	height: 32,
 	display: 'inline-block',
 	position: 'relative',
 	verticalAlign: 'top'
 };
 
-const mrMouse= [ MouseState.ENTER,MouseState.EXIT, MouseState.MOVE, MouseState.DOWN , MouseState.CLICK];
 const EMPTY= <div style={rS}></div>;
-export function MouseReadout({plotView:pv,mouseState}) {
+const EMPTY_READOUT='';
+const magMouse= [MouseState.DRAG_COMPONENT, MouseState.DRAG, MouseState.MOVE, MouseState.DOWN];
+const coordinateMap = {
+	galactic:CoordinateSys.GALACTIC,
+	eqb1950:CoordinateSys.EQ_B1950,
+	pixelSize:CoordinateSys.PIXEL,
+	sPixelSize: CoordinateSys.SCREEN_PIXEL
+};
+const labelMap = {
+	eqj2000hms:'EQ-J2000:',
+	eqj2000DCM:'EQ-J2000:',
+	galactic:'Gal:',
+	eqb1950:'Eq-B1950:',
+	fitsIP:'Image Pixel:',
+	pixelSize:'Pixel Size:',
+	sPixelSize:'Screen Pixel Size:'
+};
+
+/**
+ *
+ * @param visRoot
+ * @param plotView
+ * @param mouseState
+ * @returns {XML}
+ * @constructor
+ */
+export function MouseReadout({visRoot, plotView, mouseState}) {
 
-	if (!pv || !mouseState) return EMPTY;
+	if (!plotView || !mouseState) return EMPTY;
 
-	var plot= primePlot(pv);
 
-	var leftColumn = {width: 200, display: 'inline-block'};
+	var plot= primePlot(plotView);
+	if (!plot) return'';
+	if (isBlankImage(plot)) return EMPTY;
 
-	var rightColumn = {display: 'inline-block'};
-	var  textStyle = {textDecoration: 'underline', color: 'DarkGray', fontStyle:'italic' ,  display: 'inline-block'};
-	return (
+	if (!magMouse.includes(mouseState.mouseState)) EMPTY;
+
+	var spt= mouseState.screenPt;
+	if (!spt) return EMPTY;//hideReadout;
+
+	var column1 = {width: 90, paddingRight: 5, textAlign:'right',textDecoration: 'underline', color: 'DarkGray', fontStyle:'italic' ,  display: 'inline-block'};
+    var column2 = {width: 60, display: 'inline-block'};
+	var column3 = {width: 75, paddingRight: 5, textAlign:'right',textDecoration: 'underline', color: 'DarkGray', fontStyle:'italic' ,  display: 'inline-block'};
+	var column4 = {width: 170,display: 'inline-block'};
+	var column5 = {width: 90, paddingLeft:8, display: 'inline-block'};
+	var column5_1 = {width: 90, paddingLeft:5, display: 'inline-block'};
+
```

Comment:
Since all these column variables are not computed each time. you can move them outside the function and make them const.  That would be slightly more efficient.